### PR TITLE
make_fx can now SymIntify int inputs

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #113921
* __->__ #113452
* #113911
* #113877

This PR also contains a basket of fixes that were turned up by now testing more arguments with SymInt. I fixed as many of the easy ones as I could easily get earlier in this stack and a bunch here, but there are some more annoying ones I xfailed.

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

### CODE REVIEW

**Review by Chillee — APPROVED** [review]
test?

**Comment by ezyang** [comment]
added test

File: test/test_proxy_tensor.py [hunk 1]
```diff
@@ -1705,11 +1715,15 @@ def f(t):
...
     xfail('quantile', ''),  # Could not run 'aten::equal' with arguments from the 'Meta' backend.
-    xfail('resize_', ''),  # aten.clone.default - couldn't find symbolic meta function/decomposition
     xfail('resize_as_', ''),  # aten.clone.default - couldn't find symbolic meta function/decomposition
     xfail('unique_consecutive', ''),  # aten.unique_consecutive.default - couldn't find symbolic meta function/decomposition
     xfail('unique', ''),  # aten._unique2.default - couldn't find symbolic meta function/decomposition
 
+    xfail('linalg.norm', ''),  # AssertionError: False != True - https://github.com/pytorch/pytorch/issues/113905
+    xfail('linalg.norm', 'subgradients_at_zero'),  # https://github.com/pytorch/pytorch/issues/113905
+    xfail('max_pool2d_with_indices_backward', ''),  # Expected a value of type 'List[int]' for argument 'kernel_size' but...
+    xfail('randint_like', ''),  # when unpacking SymInt, expected int but got s0
```
Comment (ezyang) [inline test/test_proxy_tensor.py:1725]: this one is easy but involves a C++ header change, going to be next

File: test/test_proxy_tensor.py [hunk 2]
```diff
@@ -1705,11 +1715,15 @@ def f(t):
...
     xfail('nn.functional.pixel_unshuffle', ''),  # aten.pixel_unshuffle.default - couldn't find symbolic meta function/deco...
     xfail('quantile', ''),  # Could not run 'aten::equal' with arguments from the 'Meta' backend.
-    xfail('resize_', ''),  # aten.clone.default - couldn't find symbolic meta function/decomposition
     xfail('resize_as_', ''),  # aten.clone.default - couldn't find symbolic meta function/decomposition
     xfail('unique_consecutive', ''),  # aten.unique_consecutive.default - couldn't find symbolic meta function/decomposition
     xfail('unique', ''),  # aten._unique2.default - couldn't find symbolic meta function/decomposition
 
+    xfail('linalg.norm', ''),  # AssertionError: False != True - https://github.com/pytorch/pytorch/issues/113905
+    xfail('linalg.norm', 'subgradients_at_zero'),  # https://github.com/pytorch/pytorch/issues/113905
+    xfail('max_pool2d_with_indices_backward', ''),  # Expected a value of type 'List[int]' for argument 'kernel_size' but...
```
Comment (ezyang) [inline test/test_proxy_tensor.py:1724]: This one involves a fairly involved refactor of the JIT IValue parsing code

**Comment by Chillee** [comment]
You're getting failures since the testing now involves significantly more symints right?

**Comment by ezyang** [comment]
Yeah, we are now actually feeding symints for all the int arguments, as opposed to only Tensor shapes now

**Pull request closed without merging**